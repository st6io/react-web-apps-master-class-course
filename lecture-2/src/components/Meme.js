import React from 'react';
import styles from './Meme.module.scss';

const Meme = ({ topText, bottomText, image }) => {
  return (
    <div className={styles.meme}>
      <div className={styles.topText}>{topText}</div>
      <div className={styles.bottomText}>{bottomText}</div>
      <img src={image} alt="meme" className={styles.memeImage} />
    </div>
  );
};

export default Meme;
