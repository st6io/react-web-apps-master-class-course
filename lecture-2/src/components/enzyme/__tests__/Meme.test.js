import React from 'react';
import { shallow } from 'enzyme';

import Meme from '../../Meme';

describe('Meme Component', () => {
  it('smoke test', () => {
    const wrapper = shallow(<Meme className="new-class-name" />);

    expect(wrapper).toExist();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(
      <Meme
        className="new-class-name"
        topText="some text"
        bottomText="another text"
        image="https://some.test.img/jpg"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
