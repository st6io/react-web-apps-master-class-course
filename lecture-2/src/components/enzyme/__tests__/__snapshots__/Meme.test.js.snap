// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meme Component should match snapshot 1`] = `
<div
  className="meme"
>
  <div
    className="topText"
  >
    some text
  </div>
  <div
    className="bottomText"
  >
    another text
  </div>
  <img
    alt="meme"
    className="memeImage"
    src="https://some.test.img/jpg"
  />
</div>
`;
